src/main/java/picard/illumina/parser/TileMetricsUtil.java

```diff
@@ -117,18 +122,70 @@ private static File fileTileMetricsFile(File illuminaRunDirectory, Integer numCy
 
             //only create for cluster records
             if (record.isClusterRecord()) {
-                // Snag the phasing data for each read in the read structure. For both types of phasing values, this is the median of all of the individual values seen
-                final Collection<TilePhasingValue> tilePhasingValues = getTilePhasingValues(record.getLaneTileCode(),
-                        phasingMetricsFiles, readStructure, validationStringency);
-
-                tiles.add(new Tile(record.getLaneNumber(), record.getTileNumber(), tileMetricsIterator.getDensity(), record.getMetricValue(),
+                final Collection<TilePhasingValue> tilePhasingValues = phasingValues.get(record.getLaneNumber()).get(record.getTileNumber());
+                tiles.add(new Tile(record.getLaneNumber(), record.getTileNumber(), density, record.getMetricValue(),
                         tilePhasingValues.toArray(new TilePhasingValue[tilePhasingValues.size()])));
             }
         }
-
         return Collections.unmodifiableCollection(tiles);
     }
 
+    private static boolean v3TileMetricsHasClusterRecordsFirstPass(Map<String, ? extends Collection<IlluminaTileMetrics>> locationToMetricsMap) {
+        final Map.Entry<String, ? extends Collection<IlluminaTileMetrics>> entry = locationToMetricsMap.entrySet().iterator().next();
+        final IlluminaTileMetrics record = CollectionUtil.getSoleElement(entry.getValue());
+        return record.isClusterRecord();
+    }
+
+    public static Collection<Tile> parseClusterRecordsFromTileMetricsV3(
+            final Collection<File> tileMetricsOutFiles,
+            final Map<Integer, File> phasingMetricsFiles,
+            final ReadStructure readStructure
+    ) throws FileNotFoundException {
+        // Snag and cache the phasing data for each read in the read structure for each tile.
```

Comment:
keep phasingValues in memory so we don't reparse for each tile in the tileMetricsOut.bin we end up parsing. Not sure what "phasing data" is exactly, but we collect it per read per tile

Reply:
"Phasing data" is more what I meant, unfortunately

Reply:
Google is my friend! Apparently [phasing values](http://lab.loman.net/high-throughput%20sequencing/2013/11/21/diagnosing-problems-with-phasing-and-pre-phasing-on-illumina-platforms/) are some measure of lost signal due to some molecules losing sync with one another. Apparently, if it's measured, it can be corrected for. Still not 100% clear on what it means for molecules to lose sync...